crates/engine/src/types.rs

```diff
@@ -12,15 +12,29 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! Right now the `engine` crate can only be used with the `ink_env::DefaultEnvironment`.
+//!
+//! In the long-term all types in this module should be `Vec<u8>`, as to not
+//! be dependent on any specific off-chain environment type. Then the `engine`
+//! crate could be used with an arbitrary `Environment` configuration.
+//!
+//! Right now the issue is that if e.g. some test calls an `engine` API function
+//! to transfer some balance, then there has to be some `+` operation to the
+//! existing balance. And we can't just add a `Vec<u8>` onto another one, we
+//! need to know the type.
```

Comment:
This seems like an internal comment (and something that should be part of the Github issue), not something that should be in the module level comments

Reply:
Will create a follow-up once merged.